pkg/cmd/indices/config/export/export.go

```diff
@@ -0,0 +1,153 @@
+
+	indiceConfig "github.com/algolia/cli/pkg/cmd/shared/config"
+	"github.com/algolia/cli/pkg/cmdutil"
+	"github.com/algolia/cli/pkg/config"
+	"github.com/algolia/cli/pkg/iostreams"
+	"github.com/algolia/cli/pkg/utils"
+	"github.com/algolia/cli/pkg/validators"
+)
+
+type ExportOptions struct {
+	Config config.IConfig
+	IO     *iostreams.IOStreams
+
+	Indices   []string
+	Scope     []string
+	Directory string
+
+	SearchClient func() (*search.Client, error)
+}
+
+type ConfigJson struct {
+	Settings *search.Settings `json:"settings,omitempty"`
+	Rules    []search.Rule    `json:"rules,omitempty"`
+	Synonyms []search.Synonym `json:"synonyms,omitempty"`
+}
+
+// NewExportCmd creates and returns an export command for indices config
+func NewExportCmd(f *cmdutil.Factory, runF func(*ExportOptions) error) *cobra.Command {
+	opts := &ExportOptions{
+		IO:           f.IOStreams,
+		Config:       f.Config,
+		SearchClient: f.SearchClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:               "export <index>",
+		Args:              validators.ExactArgs(1),
+		ValidArgsFunction: cmdutil.IndexNames(opts.SearchClient),
+		Short:             "Export the config of one or multiple index(es)",
+		Long: heredoc.Doc(`
+			Export the config of index(es) including settings, synonyms and rules.
+		`),
+		Example: heredoc.Doc(`
+			# Export the config of the index 'TEST_PRODUCTS' to a .json in the current folder
+			$ algolia indices config export TEST_PRODUCTS
+
+			# Export the config of indexes 'TEST_PRODUCTS_1', 'TEST_PRODUCTS_2' and 'TEST_PRODUCTS_3' to a .json in the current folder
+			$ algolia indices config export TEST_PRODUCTS_1,TEST_PRODUCTS_2,TEST_PRODUCTS_3
+
+			# Export the synonyms and rules of the index 'TEST_PRODUCTS' to a .json in the current folder
+			$ algolia indices config export TEST_PRODUCTS --scope synonyms,rules
+
+			# Export the config of the index 'TEXT_PRODUCTS' to a .json into 'exports' folder
+			$ algolia indices config export TEST_PRODUCTS --directory exports
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.Indices = utils.StringToSlice(args[0])
+
+			return runExportCmd(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.Directory, "directory", "d", "", "Directory path of the output file (directy must exist)")
+	cmd.Flags().StringSliceVarP(&opts.Scope, "scope", "s", []string{}, "Scope to copy (default: all)")
+	_ = cmd.RegisterFlagCompletionFunc("scope",
+		cmdutil.StringSliceCompletionFunc(map[string]string{
+			"settings": "copy only the settings",
+			"synonyms": "copy only the synonyms",
+			"rules":    "copy only the rules",
+		}))
+
+	return cmd
+}
+
+func runExportCmd(opts *ExportOptions) error {
+	cs := opts.IO.ColorScheme()
+	client, err := opts.SearchClient()
+	if err != nil {
+		return err
+	}
+
+	for _, indexName := range opts.Indices {
+		indice := client.InitIndex(indexName)
+		hasScope := len(opts.Scope) > 0
+		var configJson ConfigJson
+
+		if !hasScope || utils.Contains(opts.Scope, "synonyms") {
```

Comment:
If you make the default `scope` flag to `[]string{"settings","synonyms","rules"}`, you won't have to repeat the `!hasScope` multiple times below.